presto-spi/src/main/java/io/prestosql/spi/function/AggregateFunction.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.spi.function;
+
+import io.prestosql.spi.block.SortOrder;
+import io.prestosql.spi.expression.ConnectorExpression;
+import io.prestosql.spi.type.Type;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.StringJoiner;
+
+import static java.util.Objects.requireNonNull;
+
+public class AggregateFunction
+{
+    // Ideally we would want this to actually be a reference to something like AggregationImplementation or aggregationFunction registry so connectors
+    // can get a one to one mapping from presto's supported aggregate function rather than relying on the name, args and output types.
+    // Something to check with Martin what he thinks is the best course here, for now just exposing the fields that the connector can use to derive if
+    // this is an agg function it can handle and information they need to push it down.
```

Comment:
Make sure to reword (or remove) before the PR is merged.

Reply:
removed.